### PR TITLE
sherman adagram 

### PR DESCRIPTION
**Pull request by Peacegypsy**
# JS Adagrams
Congratulations! You're submitting your assignment!

## Comprehension Questions
Question | Answer
--- | ---
What patterns were you able to use from your Ruby knowledge to apply to JavaScript? | iterating through arrays/hashes, case/switch statements.
Did you need to use different strategies to find information online about JavaScript than you would for Ruby? | different specificity, similar tactics.
What was a challenge you were able to overcome on this assignment? | finding new answers to challenges i had in ruby with this project, like working with hashes and arrays and iterating through them. 
What is your favorite thing about learning a new programming language? | When something runs correctly or when you understand what to write to make it work.
What is your least favorite thing about learning a new programming language? | breaking habits or patterns built in Ruby and replacing them with new ones.
Do you have any recommendations on how we could improve this project for the next cohort? | let us know a few more functions before we get started, or a recommended resource for looking them up.

### CODE REVIEW

File: src/adagrams.js [hunk 1]
```diff
@@ -1,8 +1,241 @@
...
+    Z: 10
+  },
+
+  LENGTH_BONUS: 8,
+  scoreWord(word) {
+    return word
+      .toUpperCase()
+      .split("")
+      .reduce((wordScore, letter) => {
+        const letterScore = Adagrams.scoreCHART[letter];
```
Comment (CheezItMan) [inline src/adagrams.js]: I believe this should be:  `        const letterScore = Adagrams.SCORE_CHART[letter];`

Otherwise good use of `reduce` and `split`.

File: src/adagrams.js [hunk 2]
```diff
@@ -1,8 +1,241 @@
...
+    //     Z: 10,
+    //
+    //     LENGTH_BONUS: 8
+    //   };
+
+    const hand = [];
+    let tilesDrawn = 0;
+
+    while (tilesDrawn < 10) {
+      let rand = letterPool[Math.floor(Math.random() * letterPool.length)];
```
Comment (CheezItMan) [inline src/adagrams.js:138]: I'd prefer `const`

File: src/adagrams.js [hunk 3]
```diff
@@ -1,8 +1,241 @@
...
+  //       delete drawnCopy[index];
+  //       return true;
+  //     });
+  //   }
+  // };
+  usesAvailableLetters(input, hand) {
+    const availableLetters = input.split("");
+    const handCopy = hand.slice(0);
+
+    return availableLetters.every(letter => {
```
Comment (CheezItMan) [inline src/adagrams.js:161]: Nice use of `.every`

**Comment by CheezItMan** [comment]
# JS Adagrams
## What We're Looking For

Feature | Feedback
--- | ---
**General** |
Answered comprehension questions |  Check, for JS functions, checking out the MDN is a good start, or using Google to search for solutions.
Small commits with meaningful commit messages |  Good commit messages and good number of commits
**Code Requirements** |
`drawLetters` method |  Check
Uses appropriate data structure to store the letter distribution |  Check
All tests for `drawLetters` pass |  Check
`usesAvailableLetters` method |  Check
All tests for `usesAvailableLetters` pass |  Check
`scoreWord` method | Check, with a typo, see my notes.  Otherwise it's really nicely done.
Uses appropriate data structure to store the letter scores |  Check
All tests for `scoreWord` pass |  With the typo fixed, yes.
`highestScoreFrom` method |  **MISSING**
Appropriately handles edge cases for tie-breaking logic |  **MISSING**
All tests for `highestScoreFrom` pass |  **MISSING**
**Overall** |  You hit **most** of the learning goals for the project.  The `scoreWord` method is very well designed.  Lets sit down this week and review the project, and see how the remaining methods could be done.

**Comment by Peacegypsy** [comment]
Hi Chris,
I added my highestScore function. Can you check that its okay please?

-And thank you for pointing out the typo! I’ve been trying to translate my original version in javascript and I get my ’names’ mixed up.

Christina
